Source/DafnyCore/Compilers/Dafny/DafnyBackend.cs

```diff
@@ -41,13 +41,68 @@ public class DafnyBackend : ExecutableBackend {
    * The Dafny backend is different from the other backends in that the output code needs to be compiled
    * by the Dafny compiler itself to another backend for execution.
    */
-  public override bool RunTargetProgram(string dafnyProgramName, string targetProgramText, string /*?*/ callToMain,
-    string targetFilename, ReadOnlyCollection<string> otherFileNames, object compilationResult, TextWriter outputWriter) {
+  public override bool RunTargetProgram(string dafnyProgramName, string targetProgramText, string callToMain, /*?*/
+    string targetFilename, ReadOnlyCollection<string> otherFileNames, object compilationResult, TextWriter outputWriter,
+    TextWriter errorWriter) {
     Contract.Requires(targetFilename != null || otherFileNames.Count == 0);
 
-    return RunTargetDafnyProgram(targetFilename, outputWriter);
+    /*
```

Comment:
Bad merge, the code below is in `RunTargetDafnyProgram` now, which is also used by the `LibraryBackend`. Doesn't look like you had to change the code though? It's just fine if it's still using a subprocess.

Reply:
Already on master, this code doesn't work for library back-end btw. It calls Dafny using the old CLI syntax and passing a .doo file, which causes it to crash on DooFile.Validate.

Reply:
I got that fixed FYI: https://github.com/dafny-lang/dafny/pull/3967